src/models/Student.model.js

```diff
@@ -1,32 +1,65 @@
 const Sequelize = require('sequelize-views-support');
 const sequelize = require('./sequelize.js');
 
-
-const Student = sequelize.define('students', {
-  username: {
-    type: Sequelize.STRING,
-    unique: true,
-    allowNull: false,
-  },
-  password: {
-    type: Sequelize.STRING,
-    allowNull: false,
-  },
-  age: {
-    type: Sequelize.INTEGER,
-    allowNull: false,
-  },
-  gender: {
-    type: Sequelize.BOOLEAN,
-    allowNull: false,
-  },
-  english: {
-    type: Sequelize.BOOLEAN,
-    allowNull: false,
-  },
-  postcode: {
-    type: Sequelize.STRING,
-    allowNull: false,
+const Student = sequelize.define(
+  'students',
+  {
+    username: {
+      type: Sequelize.STRING,
+      allowNull: false,
+      validate: {
+        fn(val) {
+          if (val === '') throw new Error('Please insert your username');
+        },
+        isUnique(value, next) {
+          Student.find({
+            where: {
+              username: value,
+            },
+          }).then((result) => {
+            if (result === null) {
+              return next();
+            }
+            return next(' Username already exist');
+          }).catch(err => next(err));
+        },
+      },
+    },
+    password: {
+      type: Sequelize.STRING,
+      allowNull: false,
+      validate: {
+        len: { args: 6, msg: 'Password must be at least 6 characters' },
+      },
+    },
+    age: {
+      type: Sequelize.INTEGER,
+      allowNull: false,
+    },
+    gender: {
+      type: Sequelize.BOOLEAN,
+      allowNull: false,
+      defaultValue: false,
+      validate: {
+        isBoolean: true,
+      },
+    },
+    english: {
+      type: Sequelize.BOOLEAN,
+      allowNull: false,
+      defaultValue: false,
```

Comment:
I don't think that you need to set a default value if `allowNull` is false! Remove it from all these fields.